src/hooks/UrqlUseSubscriptionWithHandler.re

```diff
@@ -1,38 +1,42 @@
+type handler('a, 'b, 'c) =
+  | Handler((option('a), 'b) => 'a): handler('a, 'b, 'a)
+  | NoHandler: handler(_, 'b, 'b);
+
 [@bs.deriving abstract]
-type useSubscriptionWithHandlerArgs = {
+type useSubscriptionArgs = {
   query: string,
   [@bs.optional]
   variables: Js.Json.t,
 };
 
 [@bs.deriving abstract]
-type useSubscriptionWithHandlerResponseJs('acc) = {
+type useSubscriptionResponseJs('a) = {
   fetching: bool,
   [@bs.optional]
-  data: 'acc,
+  data: 'a,
   [@bs.optional]
   error: UrqlCombinedError.t,
 };
 
-type useSubscriptionWithHandlerResponse('acc) = {
+type useSubscriptionResponse('a) = {
   fetching: bool,
-  data: option('acc),
+  data: option('a),
   error: option(UrqlCombinedError.t),
-  response: UrqlTypes.response('acc),
+  response: UrqlTypes.response('a),
 };
 
 [@bs.module "urql"]
-external useSubscriptionWithHandlerJs:
+external useSubscriptionJs:
   (
-    useSubscriptionWithHandlerArgs,
-    UrqlTypes.parsedHandler('acc, 'response)
+    useSubscriptionArgs,
+    option((option('a), 'b) => 'a)
   ) =>
-  array(useSubscriptionWithHandlerResponseJs('acc)) =
+  array(useSubscriptionResponseJs('c)) =
```

Comment:
This increases the type unsafety surface area, but it's safely wrapped in `useSubscription`.
In my opinion, the rationale going here can be better explained in comments if we choose this approach.